### PR TITLE
Manifest version

### PR DESCRIPTION
**Pull request by aastein**
fixes: #208 

Add Version fields to
CreateDeploymentTx
Deployment

Set Version during CreateDeployment command
Check version and log error during provider ManifestReceived stage.

### CODE REVIEW

File: sdl/sdl.go [hunk 1]
```diff
@@ -25,4 +25,4 @@ func Read(buf []byte) (SDL, error) {
 	// TODO: handle versions
 	obj := &v1{}
 	return obj, yaml.Unmarshal(buf, obj)
-}
+}
```
Comment (golangci) [inline sdl/sdl.go]: File is not goimports-ed

File: manifest/marshal.go [hunk 2]
```diff
@@ -21,3 +21,17 @@ func marshalRequest(obj *types.ManifestRequest) ([]byte, error) {
 	}
 	return buf.Bytes(), nil
 }
+
+func unmarshal(r io.Reader) (*types.Manifest, error) {
+	obj := &types.Manifest{}
+	return obj, jsonpb.Unmarshal(r, obj)
+}
+
+func marshal(obj *types.Manifest) ([]byte, error) {
```
Comment (golangci) [inline manifest/marshal.go]: obj can be github.com/ovrclk/akash/vendor/github.com/gogo/protobuf/proto.Message

File: manifest/manifest_test.go [hunk 3]
```diff
@@ -176,18 +176,68 @@ func TestVerifyRequest(t *testing.T) {
...
+
+	mani, err := sdl.Manifest()
+	require.NoError(t, err)
+
+	hash, err := Hash(mani)
+	require.NoError(t, err)
+
+	otherHash, err := Hash(&types.Manifest{
+		Groups: []*types.ManifestGroup{
+			&types.ManifestGroup{
```
Comment (golangci) [inline manifest/manifest_test.go]: File is not gofmt-ed with -s

File: manifest/hash.go [hunk 4]
```diff
@@ -0,0 +1,32 @@
+package manifest
+
+import (
+	"bytes"
+	"crypto/md5"
```
Comment (golangci) [inline manifest/hash.go]: G501: Blacklisted import crypto/md5: weak cryptographic primitive

File: manifest/hash.go [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+)
+
+var ErrDifferentHashes = errors.New("manifest hash does not match the expected hash")
+
+func Hash(m *types.Manifest) ([]byte, error) {
+	bytes, err := marshal(m)
+	if err != nil {
+		return nil, err
+	}
+	h := md5.New()
```
Comment (golangci) [inline manifest/hash.go]: G401: Use of weak cryptographic primitive

File: manifest/marshal.go [hunk 6]
```diff
@@ -21,3 +21,17 @@ func marshalRequest(obj *types.ManifestRequest) ([]byte, error) {
 	}
 	return buf.Bytes(), nil
 }
+
+func unmarshal(r io.Reader) (*types.Manifest, error) {
```
Comment (golangci) [inline manifest/marshal.go]: `unmarshal` is unused

File: manifest/hash.go [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+	h.Write(bytes)
```
Comment (golangci) [inline manifest/hash.go]: Error return value of `h.Write` is not checked

File: manifest/hash.go [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+	return h.Sum(nil), err
+}
+
+func verifyHash(m *types.Manifest, expectedHash []byte) error {
+	hash, err := Hash(m)
+	if err != nil {
+		return err
+	}
+	if bytes.Compare(hash, expectedHash) != 0 {
```
Comment (golangci) [inline manifest/hash.go]: should use !bytes.Equal(hash, expectedHash) instead

File: manifest/marshal.go [hunk 9]
```diff
@@ -21,3 +21,12 @@ func marshalRequest(obj *types.ManifestRequest) ([]byte, error) {
 	}
 	return buf.Bytes(), nil
 }
+
+func marshal(obj *types.Manifest) ([]byte, error) {
```
Comment (golangci) [inline manifest/marshal.go]: obj can be github.com/ovrclk/akash/vendor/github.com/gogo/protobuf/proto.Message

File: manifest/hash.go [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+	"bytes"
+	"crypto/sha256"
+	"errors"
+
+	"github.com/ovrclk/akash/types"
+)
+
+var ErrDifferentHashes = errors.New("manifest hash does not match the expected hash")
+
+func Hash(m *types.Manifest) ([]byte, error) {
```
Comment (golangci) [inline manifest/hash.go]: m can be github.com/ovrclk/akash/vendor/github.com/gogo/protobuf/proto.Message

File: manifest/hash.go [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+	}
+	h := sha256.New()
+	_, err = h.Write(bytes)
+	if err != nil {
+		return nil, err
+	}
+	return h.Sum(nil), err
+}
+
+func verifyHash(m *types.Manifest, expectedHash []byte) error {
```
Comment (golangci) [inline manifest/hash.go]: m can be github.com/ovrclk/akash/vendor/github.com/gogo/protobuf/proto.Message

Comment (aastein) [reply manifest/hash.go]: I'm gonna stop linting here and do another pr that covers the entire project

Comment (boz) [reply manifest/hash.go]: I'm fine with just turning it off for now.  Just wanted to try it out.

Don't bother w/ fixing issues unless you think they're reasonable & it won't make the codebase too inconsistent.

File: types/types.proto [hunk 12]
```diff
@@ -161,7 +162,8 @@ message TxCreateDeployment {
   bytes  tenant  = 1 [(gogoproto.customtype)="github.com/ovrclk/akash/types/base.Bytes",(gogoproto.nullable) = false];
   uint64 nonce = 2;
   int64 orderTTL = 3;
-  repeated GroupSpec groups = 4;
+  bytes version = 4;
```
Comment (boz) [inline types/types.proto]: `[(gogoproto.customtype)="github.com/ovrclk/akash/types/base.Bytes",(gogoproto.nullable) = false]`?

Comment (aastein) [reply types/types.proto]: I thought base.Bytes is only relevant for addressing?

**Pull request merged**